Sources/DatadogObjc/DDURLSessionDelegate+objc.swift

```diff
@@ -8,4 +8,42 @@ import Foundation
 import Datadog
 
 @objc
-public class DDNSURLSessionDelegate: DDURLSessionDelegate {}
+open class DDNSURLSessionDelegate: NSObject, URLSessionTaskDelegate {
```

Comment:
There's a bunch of critical and not very obvious implementations in this class - this needs tests to document and assert expected behaviours.

Reply:
`TracingNSURLSessionScenario` and `RUMNSURLSessionResourcesScenario` test `DDNSURLSessionDelegate`, what other test cases may we need?

what about that @ncreated ?
```
/// NOTE: RUMM-1234
/// `DDNSURLSessionDelegate` imitates `DDURLSessionDelegate` class
/// it responds to whatever `DDURLSessionDelegate` responds to and
/// it forwards every method call to its private `DDURLSessionDelegate` instance.
/// That way, `DDNSURLSessionDelegate` inherits `DDURLSessionDelegate` functionality
/// without actual inheritance/superclass.
```

Reply:
There are no `DDNSURLSessionDelegate` unit tests - there was no point having it as this class was only subclassing the swift delegate. Now that it executes its own logic we need to add tests as usual.

Tests you referred to are integration tests covering basic scenarios. We can't leverage them for documenting the code, but we can do it easily with few unit tests.